src/core/RenderContext.cpp

```diff
@@ -130,17 +128,23 @@ void RenderContext::updateGLWindows()
 
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
+        window->setBlockDrawCalls( false );
         window->viewport()->repaint();
+        window->setBlockDrawCalls( true );
         ++activeGLWindowIndex_;
     }
+    glFinish();
 }
 
 void RenderContext::swapBuffers()
 {
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
-        static_cast<QGLWidget*>( window->viewport( ))->makeCurrent();
-        static_cast<QGLWidget*>( window->viewport( ))->swapBuffers();
+        QGLWidget* glContext = static_cast<QGLWidget*>( window->viewport( ));
+        glContext->makeCurrent();
+        if( window != windows_.front( ))
+            window->disableVSync();
```

Comment:
it's enough to do this once imo, not every swap.See https://github.com/Eyescale/Equalizer/blob/master/eq/client/glx/window.cpp:124